app/src/main/java/com/arygm/quickfix/ui/elements/ScrollableRow.kt

```diff
@@ -0,0 +1,132 @@
+              .aspectRatio(0.9f) // Maintain a consistent aspect ratio for height
+              // .fillMaxHeight()
+              .shadow(elevation = 5.dp, shape = RoundedCornerShape(8.dp))
+              .testTag("ServiceCard_${service.name}"),
+      shape = RoundedCornerShape(8.dp),
+      colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
+      onClick = onClick) {
+        Column(
+            modifier = Modifier.fillMaxSize(), horizontalAlignment = Alignment.CenterHorizontally) {
+              Image(
+                  painter = painterResource(id = service.imageResId),
+                  contentDescription = service.name,
+                  modifier =
+                      Modifier.fillMaxWidth()
+                          .weight(0.7f) // Allocate 70% of card height to the image
+                          .testTag("ServiceImage_${service.name}"),
+                  contentScale = androidx.compose.ui.layout.ContentScale.Crop)
+              Text(
+                  text = service.name,
+                  color = MaterialTheme.colorScheme.onBackground,
+                  textAlign = TextAlign.Center,
+                  style = MaterialTheme.typography.headlineMedium,
+                  maxLines = 1,
+                  modifier =
+                      Modifier.fillMaxWidth()
+                          .weight(0.3f) // Allocate 30% of card height to the text
+                          .wrapContentHeight(Alignment.CenterVertically)
+                          .testTag("ServiceName_${service.name}"))
+            }
+      }
+}
+
+// PopularServicesRow composable with percentage-based card width and consistent padding
+@Composable
+fun PopularServicesRow(
+    services: List<Service>,
+    onServiceClick: (Service) -> Unit,
+    modifier: Modifier = Modifier
+) {
+  BoxWithConstraints {
+    val cardWidth = maxWidth * 0.4f // 40% of the available width for each card
+    val horizontalSpacing = maxWidth * 0.025f // 2.5% of the available width for spacing
+
+    LazyRow(
+        modifier = modifier.fillMaxWidth().testTag("PopularServicesRow"),
+        horizontalArrangement = Arrangement.spacedBy(horizontalSpacing),
+        contentPadding = PaddingValues(start = horizontalSpacing, end = horizontalSpacing)) {
+          items(services.size) { index ->
+            ServiceCard(service = services[index], modifier = Modifier.width(cardWidth)) {
+              onServiceClick(services[index])
+            }
+          }
+        }
+  }
+}
+
+// MainContent composable (the screen that contains all sections)
+@Composable
+fun MainContent() {
+  val backgroundColor = MaterialTheme.colorScheme.background
+
+  val services =
+      listOf(
+          Service("Mechanic", R.drawable.mechanic),
+          Service("Gardener", R.drawable.gardener),
+          Service("Electrician", R.drawable.electrician),
+          Service("Painter", R.drawable.painter),
+          Service("Plumber", R.drawable.plumber),
```

Comment:
Why are only 3 services shown ?

Reply:
You need to click on 'Show More' to see all the services.